api/app/Generators/PoolCandidateCsvGenerator.php

```diff
@@ -90,133 +94,123 @@ public function generate(): self
         $this->spreadsheet = new Spreadsheet();
 
         $sheet = $this->spreadsheet->getActiveSheet();
+
+        $currentCandidate = 1;
+        $query = $this->buildQuery();
+        $query->chunk(200, function ($candidates) use ($sheet, &$currentCandidate) {
+            foreach ($candidates as $candidate) {
+
+                if (! in_array($candidate->pool_id, $this->poolIds)) {
```

Comment:
Could you please add a comment for what this special handling represents?  It's not really clear to me.

Reply:
For sure, hopefully this makes sense :sweat_smile: 

[document pool ids push](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11300/commits/60e6f1aaa5af23aab3b4b9433567a31a9d52a4c0)